hail/src/test/scala/is/hail/expr/types/physical/PContainerTest.scala

```diff
@@ -3,63 +3,177 @@ package is.hail.expr.types.physical
+
+    val res = fb.result()()(src)
+    res
+  }
+
+  def testHasMissingValues(sourceType: PArray, data: IndexedSeq[Any]) = {
+    val srcRegion = Region()
+    val src = ScalaToRegionValue(srcRegion, sourceType, data)
+
+    log.info(s"\nTesting $data")
+
+    val fb = EmitFunctionBuilder[Long, Boolean]("not_empty")
+    val value = fb.getArg[Long](1)
+
+    fb.emit(sourceType.hasMissingValues(value))
+
+    val res = fb.result()()(src)
+    res
+  }
+
+  @Test def checkFirstNonZeroByte() {
+    val sourceType = PArray(PInt64(false))
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(0, 0)) == false)
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(1, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(1, 1)) == true)
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(8, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(8, 1)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(8, 8)) == true)
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(32, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(31, 31)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(32, 32)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(33, 33)) == true)
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(64, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(64, 1)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(64, 32)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(64, 33)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(64, 64)) == true)
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(68, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(68, 1)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(68, 32)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(68, 33)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(68, 64)) == true)
 
+    assert(testContainsNonZeroBits(sourceType, nullInByte(72, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(72, 1)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(72, 32)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(72, 33)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(72, 64)) == true)
+
+    assert(testContainsNonZeroBits(sourceType, nullInByte(73, 0)) == false)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(73, 1)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(73, 32)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(73, 33)) == true)
+    assert(testContainsNonZeroBits(sourceType, nullInByte(73, 64)) == true)
+  }
+
+  @Test def checkFirstNonZeroByteStaged() {
```

Comment:
I removed most of these tests, probably not necessary, since they duplicate unstated code, enough to show the staged indirection works in a true and false case.